src/sql/util.py

```diff
@@ -556,11 +562,14 @@ def is_non_sqlalchemy_error(error):
         "pyodbc.ProgrammingError",
         # Clickhouse errors
         "DB::Exception:",
-        # Pyspark
-        "UNRESOLVED_ROUTINE",
-        "PARSE_SYNTAX_ERROR",
     ]
-    return any(msg in str(error) for msg in specific_db_errors)
+    is_pyspark_analysis_exception = (
+        isinstance(error, AnalysisException) if AnalysisException else False
+    )
+    return (
+        any(msg in str(error) for msg in specific_db_errors)
+        or is_pyspark_analysis_exception
+    )
```

Comment:
If AnalysisException is imported then checks if the error is of instance of pyspark's Analysis Exception and handles it accordingly